Assets/MRTK/Core/Definitions/Devices/DeviceInputType.cs

```diff
@@ -29,11 +29,22 @@ public enum DeviceInputType
         ButtonPress,
         ButtonTouch,
         ButtonNearTouch,
+        PrimaryButtonPress,
```

Comment:
Is it safe to add new values in the middle of the enum? i.e. is this serialized as value (i.e. integer) in any location in any of our .assets or any customer .assets?

A quick search doesn't show anything.

If we wanted to be careful with how we update enums we should be adding to the end to avoid this class of issues (this a generally reasonable practice anywhere you go)

Reply:
Ah wait I see all of the serialized asset changes

Reply:
Combining my comment with this thread:

>Adding these to the enum will likely break all existing profiles, since this is serialized into the controller mapping profile and this will change the underlying values of all following entries (like Trigger below).
>
>Although I agree that this placement makes the most sense to locate them near related entries, we should add them to the end to maintain existing enum values (or mark each one explicitly, but that could get messy).

Reply:
Although our assets might be updated, this may break any custom profiles that aren't re-serialized.

Though...I've been thinking about ignoring the serialized values in many cases, since it's not editable by a user and we should just use the "latest" values from the class itself.

Reply:
fixed things up!